[4.0] Redo the com_templates

dgrammatiko opened the pull request:
Pull Request for Issue # .


### Summary of Changes

One view to rule them all.
- Removes the 2 different entry layouts for the templates (template, style)
- Introduces the styling from Konshu template
- Fixes all the menus
- Brings the install logic from Konshu

This PR largely is the adoption of the work that was done by @kawshar and his team in their backend template. So all the kudos should go to them, I'm just bringing their code to the current branch (with their permissions)

### Testing Instructions

This is part of a wider rewrite of com_templates, probably has some a11y issues but should be functionally on par with the previous state (actually should be better as everything is one click away)

### Actual result BEFORE applying this Pull Request


### Expected result AFTER applying this Pull Request

![Screenshot 2021-03-28 at 17 39 31](https://user-images.githubusercontent.com/3889375/112758142-15299580-8fed-11eb-99f0-48cd5952b32c.png)

![Screenshot 2021-03-28 at 17 39 48](https://user-images.githubusercontent.com/3889375/112758153-1c50a380-8fed-11eb-9301-ec6e0730b87c.png)

![Screenshot 2021-03-28 at 17 43 45](https://user-images.githubusercontent.com/3889375/112758170-2b375600-8fed-11eb-95db-ef9769f8f9ee.png)

Install template with one click
![Screenshot 2021-03-28 at 19 00 28](https://user-images.githubusercontent.com/3889375/112760622-03012480-8ff8-11eb-8007-4854496ae307.png)

![Screenshot 2021-03-28 at 19 00 40](https://user-images.githubusercontent.com/3889375/112760629-08f70580-8ff8-11eb-9a09-33be5fa9e9f7.png)


### Documentation Changes Required

@brianteeman can you review the a11y and the strings here?

brianteeman commented on administrator/components/com_templates/src/Model/StylesModel.php:
```diff
@@ -11,12 +11,14 @@
 use Joomla\CMS\MVC\Model\ListModel;
 use Joomla\Database\ParameterType;
-use Joomla\String\StringHelper;
+use Joomla\CMS\Uri\Uri;
```
alpha sort

dgrammatiko committed:
alpha sort

dgrammatiko committed:
meh

brianteeman commented:
looks good. will test properly later

dgrammatiko committed:
oops

dgrammatiko committed:
Add install template logic

brianteeman commented on administrator/components/com_templates/tmpl/styles/default_modal_install.php:
```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Installer.packageinstaller
+ *
+ * @copyright   Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
```
copy paste error

dgrammatiko replied:
fixed

brianteeman commented on administrator/language/en-GB/com_templates.ini:
```diff
@@ -253,3 +253,7 @@ COM_TEMPLATES_STYLE_SET_DEFAULT="Set as Default"
 COM_TEMPLATES_GRID_UNSET_LANGUAGE="Unset Language"
 COM_TEMPLATES_STYLES_PAGES_ALL_LANGUAGE="All"
+JTOOLBAR_INSTALL_TEMPLATE="Install template"
+PLG_INSTALLER_PACKAGEINSTALLER_UPLOAD_INSTALL_JOOMLA_EXTENSION="Install a new template"
```
i know its still a draft but ....

dgrammatiko replied:
@brianteeman let's agree on the structure and the features first and then I will clean up all this mess

dgrammatiko replied:
fixed

Quy commented:
Go to Home Dashboard.
Click Cog icon under Site and select Edit.
Remove `Template Code Icon` option.

dgrammatiko committed:
remove the extra link

dgrammatiko commented:
@Quy done, thanks, good catch

dgrammatiko committed:
comments

dgrammatiko committed:
CS

dgrammatiko committed:
wow

Bond97 commented:
I dont know, should I create new issue, but I write my suggestion in this PR.

When I upgrade Joomla 4.0.0 beta8-dev to new daily version (check updates and reinstall Joomla core files), sometimes there is a notification "%number% overriden files changed" and in this frame text changed to "%number% overrides to check".
![image](https://user-images.githubusercontent.com/78050514/112762384-07353e00-9008-11eb-8cd4-4d415a19c6ac.png)

I go to administrator/index.php?option=com_templates&view=templates, in override files column show yellow "changes found". Usually it happens with helixultimate template (new 2.0.0 beta version). With cassiopeia that notification not showing.

My suggestion: show, which of files has changed or overriden. Maybe it be useful for @dgrammatiko attempt to rewrite com_template.

Or it's unnecessary function?

brianteeman commented:
The main change I would make is the way that multiple styles of the same template are handled.

In khansu they were all listed together but in this version they duplicate all the other template stuff.

![image](https://user-images.githubusercontent.com/1296369/112762338-09db6780-8ff7-11eb-8b74-0a44fe8f8795.png)

Plus the preview is for the style not the template. So I would be looking to make the layout more like
![image](https://user-images.githubusercontent.com/1296369/112762689-e6b1b780-8ff8-11eb-971d-58a53f306ecf.png)

brianteeman commented:
> My suggestion: show, which of files has changed or overriden. 

And if there are 25 changed files?

dgrammatiko commented:
> Plus the preview is for the style not the template. So I would be looking to make the layout more like

That was easy

> In khansu they were all listed together but in this version they duplicate all the other template stuff.

This needs some query tweaking

Bond97 commented:
> And if there are 25 changed files?
 I don't know about that, it's just my suggestion :) I think, when number of changed files are more than 15-20, it's hard to read list all of them.
 
 Maybe it's unnesessary function.

brianteeman commented:
@dgrammatiko I think you also need to rethink the actions menu - it might not even be needed - so that its clear if the action is for the style or the template

ceford commented:
Looks good. The Title bar might be Templates: Site or Templates Administrator as required. I think the default template should no be 'deletable'. And I guess the double icons for Info and Preview will get fixed. The first screen grab shows a different side menu order - I assume that is a user choice and not an intended default (no need to answer that unless it is).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32896">issues.joomla.org/tracker/joomla-cms/32896</a>.</sub>

kawshar commented:
Why don't you use the exact UX that we proposed?

dgrammatiko commented:
@kawshar I think I kept it almost intact (apart from changing some elements for a11y and using the existing styling, not going to restyle everything here)

![Screenshot 2021-03-29 at 17 13 57](https://user-images.githubusercontent.com/3889375/112858745-72d3e580-90b2-11eb-8126-2a46a609be07.png)
![Screenshot 2021-03-29 at 17 14 44](https://user-images.githubusercontent.com/3889375/112858766-79625d00-90b2-11eb-84f5-c8d68c2d220e.png)

ceford commented:
Another thought: I wonder if `Edit Files` and `Edit Style` might be simplified to `Files` and `Options`? For me the word Style leads me to expect to do something with css but that is done by editing user.css.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32896">issues.joomla.org/tracker/joomla-cms/32896</a>.</sub>

brianteeman commented:
@ceford that does indeed make sense when you only have one style. As soon as you have multiple styles it becomes more confusing.

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+
+$wa->useStyle('com_templates.admin-styles');
 ?>
+<style>
```
do you really need this?

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+													<span class="template-version small text-muted ml-2">v<?php echo $this->escape($version); ?></span>
+												<?php endif; ?>
+												<?php if (($item->home != '0' && $item->home != '1') &&  $item->image):?>
+													<span class="ml-2">
```
```suggestion
													<span class="ms-2">
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+													<?php endif; ?>
+												</span>
+												<?php if ($version = $item->xmldata->get('version')) : ?>
+													<span class="template-version small text-muted ml-2">v<?php echo $this->escape($version); ?></span>
```
```suggestion
													<span class="template-version small text-muted ms-2">v<?php echo $this->escape($version); ?></span>
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+														<?php echo HTMLHelper::_('image', 'mod_languages/' . $item->image . '.gif', $item->language_title, array('title' => Text::sprintf('COM_TEMPLATES_STYLES_PAGES_ALL_LANGUAGE', $this->escape($item->language_title))), true); ?>
+													</span>
+												<?php elseif ($item->assigned > 0) : ?>
+													<span class="ml-2">
```
```suggestion
													<span class="ms-2">
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+												</div>
+												<?php if ($clientId === 0) : ?>
+												<a href="<?php echo Route::_( Uri::root() . 'index.php?tp=1&templateStyle=' . (int) $item->id); ?>" target="_blank" class="card-header-icon" title="<?php echo Text::_('COM_TEMPLATES_PREVIEW'); ?>">
+													<span class="icon-eye-open icon-md" area-hidden="true"></span>
```
```suggestion
													<span class="icon-eye-open icon-md" aria-hidden="true"></span>
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+											<div class="list-group-item">
+												<div class="d-flex">
+													<?php if($canCreate  || $canDelete) : ?>
+													<div class="mr-auto">
```
```suggestion
													<div class="me-auto">
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+													<?php if($canCreate  || $canDelete) : ?>
+													<div class="mr-auto">
+														<?php if($canCreate) : ?>
+														<a class="btn btn-link" href="javascript:void(0);" onclick="return Joomla.listItemTask('cb<?php echo $i; ?>', 'styles.duplicate')">
```
This should be a button not a link pretending to be a button

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+														<?php endif; ?>
 
+														<?php if($canDelete) : ?>
+														<a class="btn btn-link" href="javascript:void(0);" onclick="return Joomla.listItemTask('cb<?php echo $i; ?>', 'styles.delete')">
```
This should be a button not a link pretending to be a button

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+														<a href="<?php echo Route::_('index.php?option=com_templates&view=template&id=' . (int) $item->e_id); ?>" class="btn btn-link">
+															<?php echo Text::_('COM_TEMPLATES_TEMPLATE_EDIT_FILES'); ?>
+														</a>
+														<a href="<?php echo Route::_('index.php?option=com_templates&task=style.edit&id=' . (int) $item->id); ?>" title="<?php echo Text::_('JACTION_EDIT'); ?> <?php echo $this->escape(addslashes($item->title)); ?>" class="btn btn-link ml-3">
```
```suggestion
														<a href="<?php echo Route::_('index.php?option=com_templates&task=style.edit&id=' . (int) $item->id); ?>" title="<?php echo Text::_('JACTION_EDIT'); ?> <?php echo $this->escape(addslashes($item->title)); ?>" class="btn btn-link ms-3">
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+
+											<div class="card-footer d-flex align-items-center p-4">
+												<?php if ($item->home == '0') : ?>
+													<a href="javascript:void(0);" onclick="return Joomla.listItemTask('cb<?php echo $i; ?>','styles.setDefault')" class="btn btn-secondary btn-block">
```
This should be a button not a link pretending to be a button

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+													</a>
+												<?php elseif ($item->home == '1'):?>
+													<strong class="text-success btn btn-transparent">
+														<span class="icon-check-circle icon-md mr-2" area-hidden="true"></span> <?php echo Text::_('COM_TEMPLATES_STYLE_IS_DEFAULT'); ?>
```
```suggestion
														<span class="icon-check-circle icon-md me-2" area-hidden="true"></span> <?php echo Text::_('COM_TEMPLATES_STYLE_IS_DEFAULT'); ?>
```

brianteeman commented on administrator/components/com_templates/tmpl/styles/default.php:
```diff
@@ -15,132 +15,197 @@
+													</span>
+												<?php elseif ($item->assigned > 0) : ?>
+													<span class="ml-2">
+														<span class="icon-batch duotone" area-hidden="true" title="<?php echo Text::sprintf('COM_TEMPLATES_STYLES_PAGES_SELECTED', $this->escape($item->assigned)); ?>"></span>
```
duotone ?? I think this is a copy paste error from khansu

dgrammatiko committed:
Update administrator/components/com_templates/tmpl/styles/default.php

Co-authored-by: Brian Teeman <brian@teeman.net>

dgrammatiko committed:
Update administrator/components/com_templates/tmpl/styles/default.php

Co-authored-by: Brian Teeman <brian@teeman.net>

dgrammatiko committed:
Update administrator/components/com_templates/tmpl/styles/default.php

Co-authored-by: Brian Teeman <brian@teeman.net>

dgrammatiko committed:
Update administrator/components/com_templates/tmpl/styles/default.php

Co-authored-by: Brian Teeman <brian@teeman.net>

dgrammatiko committed:
Update administrator/components/com_templates/tmpl/styles/default.php

Co-authored-by: Brian Teeman <brian@teeman.net>

dgrammatiko committed:
Update administrator/components/com_templates/tmpl/styles/default.php

Co-authored-by: Brian Teeman <brian@teeman.net>

ghazal commented:
I have tested this item :white_check_mark: successfully on 4f5c35d4cb7124f15c1a6145633f6342f9e69d85<br /><br />Nice changes.
I really hope it will be accepted.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32896">issues.joomla.org/tracker/joomla-cms/32896</a>.</sub>

dgrammatiko committed:
more

dgrammatiko commented:
@ghazal thanks for testing this but it's still a draft, needs some more work before it gets to the testing stage...

dgrammatiko committed:
preview per style

dgrammatiko committed:
layouts

dgrammatiko committed:
more

brianteeman committed:
Update QuickIconHelper.php

use the correct param or the icon wont show ;)

brianteeman commented:
The quickicon doesnt display. I have made a pr to your branch to fix that
https://github.com/dgrammatiko/joomla-cms/pull/1

dgrammatiko committed:
more

astridx commented:
I have just tested this branch. Thank you for this.

Can I still make a copy of a template in this PR? I can't find the button. 

And then I noticed this:

- Go to Home Dashboard.
- Click _Overrides are up to date_
- You get an 500er Error.

![Home Dashboard template Administration](https://user-images.githubusercontent.com/9974686/113421054-b38d7080-93ca-11eb-80d3-0d463f401377.png)

![Error 500](https://user-images.githubusercontent.com/9974686/113421088-c4d67d00-93ca-11eb-969d-8a3d6b9d9a04.png)

dgrammatiko commented:
> Can I still make a copy of a template in this PR? I can't find the button.

Only for Legacy templates.

> Click Overrides are up to date

I'll have to check why it's broken, bet I have to say that the overrides checks are very badly implemented, maybe I will redo them or if it is too messy will have to ask for removal and reintroduce them properly in 4.1 or later

dgrammatiko commented:
Ok an update here, there's still a bit more work for this view but I want to share this as some of the comments are already addressed or I'm currently working on them:

The template card now looks like this (addressing @brianteeman 's comment of not having multiple cards per template, just to mention that the khansu had one card per style) :

![Screenshot 2021-04-02 at 16 20 58](https://user-images.githubusercontent.com/3889375/113423952-a757e200-93cf-11eb-9595-ea178c55fe0d.png)

![Screenshot 2021-04-02 at 16 21 06](https://user-images.githubusercontent.com/3889375/113423967-ade65980-93cf-11eb-822b-062f1c8fe3a2.png)


![Screenshot 2021-04-02 at 16 21 14](https://user-images.githubusercontent.com/3889375/113423982-b474d100-93cf-11eb-9032-3f33ab5383f7.png)


Not all the actions are yet wired to the backend and also the child templates are not yet properly in the loop. In sort still WIP...

dgrammatiko committed:
oops

dgrammatiko committed:
CS

brianteeman commented:
Looking good. I would keep edit files separate from the styles actions. Otherwise keep up the great work

dgrammatiko commented:
> Looking good. I would keep edit files separate from the styles actions

Yeah, this together with the `template delete` should be on the header of the card (I guess)

brianteeman commented:
> > Looking good. I would keep edit files separate from the styles actions
> 
> Yeah, this together with the `template delete` should be on the header of the card (I guess)

Isnt it style delete?

dgrammatiko committed:
CS

dgrammatiko commented:
> Isnt it style delete?

Yes, there is a `Delete Style` but we should be able to also delete a template from this view (assuming that there's no active menu, etc). If we can pull this then from this view users could: install, delete, duplicate or create child templates and also manage all the styles. So no navigation links are required (at least this is my aim here)

brianteeman commented:
you dont delete a template. you uninstall a template

dgrammatiko commented:
> you dont delete a template. you uninstall a template

you're right install/uninstall are the right terms...

dgrammatiko committed:
Wire some actions

dgrammatiko committed:
more links

dgrammatiko committed:
x

dgrammatiko committed:
x

dgrammatiko committed:
fixes

dgrammatiko committed:
Fix the thumbnails, save inheritable, parent in the db manifest data

brianteeman commented:
@dgrammatiko what needs to be done to get this merged. __IF__ its because its waiting on the child templates is that realisticaly achievable for 4.0. If not then can this be used without child templates

dgrammatiko commented:
> @dgrammatiko what needs to be done to get this merged.

- I haven't tested the multilingual stuff
- I need to finish the new Override creator
- I need to patch the file manager in order to cope with both modes

It's not a huge amount of work but I always get distracted by ppl mention me in other issues so I switch branched to fix an easy bug but then refocusing here is hard. That said I should be able to move it out of WIP in the coming days

![Screenshot 2021-05-08 at 11 48 15](https://user-images.githubusercontent.com/3889375/117534847-6d43b680-aff3-11eb-93ce-09717bd70725.png)

brianteeman commented:
great - let me know if I can help at all to get this over the final hurdles

rinka88 commented:
when can I test it?

astridx commented:
I would also be interested to know if this function will be included in a J4 release candidate.

chmst commented:
@astridx unfortunately too late for 4.0.

astridx commented:
@chmst Thanks for the answer.

rinka88 commented:
I think we need to revive this pr. And designate which branch will be included, 4.0.x or 4.x?

dgrammatiko commented:
> I think we need to revive this pr.

The PR was never dead but there were higher priorities for J4.0 but if you want to take over please be my guest

dgrammatiko committed:
update

dgrammatiko committed:
One view to rule them all

dgrammatiko committed:
cs

dgrammatiko committed:
use the existing code (no ajax)

rinka88 commented:
@dgrammatiko I mean from wip. How to do this, become a guest?

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+ * @package     Joomla.Administrator
+ * @subpackage  com_templates
+ *
+ * @copyright   (C) 2009 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+/**
+ * Template templates list controller class.
+ *
+ * @since  1.6
```
```suggestion
 * @since  __DEPLOY_VERSION__
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+	 *
+	 * @return  BaseDatabaseModel
+	 *
+	 * @since   1.6
```
```suggestion
	 * @since   __DEPLOY_VERSION__
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+	 *
+	 * @return  void
+	 *
+	 * @since   1.6
+	 */
+	public function setDefault()
```
```suggestion
	 * @since   __DEPLOY_VERSION__
	 */
	public function setDefault()
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+	 *
+	 * @return  void
+	 *
+	 * @since   1.6
```
```suggestion
	 * @since   __DEPLOY_VERSION__
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+	 *
+	 * @return  boolean
+	 *
+	 * @since   4.0.0
```
```suggestion
	 * @since   __DEPLOY_VERSION__
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+	 *
+	 * @return  boolean     true on success, false otherwise
+	 *
+	 * @since   4.0.0
```
```suggestion
	 * @since   __DEPLOY_VERSION__
```

PhilETaylor commented on administrator/components/com_templates/src/Controller/TemplatesController.php:
```diff
@@ -0,0 +1,322 @@
+	 *
+	 * @return  void
+	 *
+	 * @since   4.0.0
```
```suggestion
	 * @since   __DEPLOY_VERSION__
```

PhilETaylor commented on administrator/components/com_templates/src/View/Templates/HtmlView.php:
```diff
@@ -3,7 +3,7 @@
  * @subpackage  com_templates
  *
- * @copyright   (C) 2008 Open Source Matters, Inc. <https://www.joomla.org>
+ * @copyright   (C) 2009 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
Update administrator/components/com_templates/src/Controller/TemplatesController.php

Co-authored-by: Phil E. Taylor <phil@phil-taylor.com>

dgrammatiko committed:
tb

dgrammatiko committed:
more

dgrammatiko committed:
zzz

dgrammatiko committed:
overrides creator init

dgrammatiko committed:
Overrides creator as a custom element

dgrammatiko commented:
Sneak preview for the overrides creator:

https://user-images.githubusercontent.com/3889375/132101328-be1643c8-e9af-4b75-8645-a9162d554214.mp4

dgrammatiko committed:
CS

dgrammatiko committed:
JS CS

dgrammatiko committed:
Strings clean up

dgrammatiko committed:
Info modal + overrides fixes

dgrammatiko committed:
JS CS

RickR2H commented:
@dgrammatiko let me know if I can help test the feature! 😉

dgrammatiko committed:
- code cleanup
- Custom names only for components,modules (no layouts, plugins as there is no way to redeclare a different layout name for them)
- Wire the backend for the creation of the file, WIP

dgrammatiko commented:
@RickR2H it's not fully functional atm (it creates just the html dir structure in the template) but if you have some time and test the front end and give some feedback that would be valuable

dgrammatiko committed:
- Fixes
- Introduces the FE for template forking/creating Child templates
- more strings
- Still WIP

dgrammatiko committed:
JS CS

dgrammatiko commented:
Sneak preview of template duplication/Child template creation

![Screenshot 2021-09-05 at 18 06 27](https://user-images.githubusercontent.com/3889375/132133736-c0de7302-898a-4503-a5c7-4f283bdcc387.png)

dgrammatiko committed:
Some clean up

dgrammatiko committed:
More intuitive template name/action

dgrammatiko committed:
JS Cleanup

dgrammatiko committed:
Meh, do not disable ESLint when we can fix the CS

dgrammatiko committed:
c

chmst commented:
It was my dream since I am using Joomla to have the two template views in one, thanks for working on that.

But I have a few remarks and think ithat the appearance should be consistent with atum.

- In the menu an Icon-arrow-down stands for installation when on all other places atum uses icon-upload
- A select list where I expect options for selection is not a select list but kind information bubble 
- What means the word Legacy here?
- A cog  the actions dropdown (@brianteeman  uses another coq for that withour dropdown arrow)
- Icons for actions are not unique (delete = uninstall, fork = new)
- <h3> for "styles" is a rupture in the structure (a11y). 
- would be nice to have the same behaviour of the default icon as in other lists.

dgrammatiko commented:
> But I have a few remarks and think ithat the appearance should be consistent with atum.

Generally speaking, this is still WIP so things will eventually be more consistent

> A select list where I expect options for selection is not a select list but kind information bubble

There are 2 commands there: send email to support and get the template description

> What means the word Legacy here?

Legacy is referring to the templates that CANNOT have child templates, Modern will be the ones that do support them 

> h3 tag for "styles" is a rupture in the structure (a11y).

The card footer is not really representative of what will be there finally, it needs work

> would be nice to have the same behaviour of the default icon as in other lists.

There's a star icon that is supposed to do that, the logic is missing atm

About all the icons, I just picked random ones but it's an easy task to change them

dgrammatiko commented:
This one could be revisited in v5

brianteeman commented:
@dgrammatiko please re-open this. Dont allow yourself to be bullied by ignorant fools (unless of course they were correct and you were paid to mess joomla up as claimed on tiwtter)

dgrammatiko commented:
@brianteeman this is replaced by 
- https://github.com/joomla/joomla-cms/pull/35874
- https://github.com/joomla/joomla-cms/pull/35998
- https://github.com/joomla/joomla-cms/pull/36011

for 4.1 according to the maintainers' guidance. This can still be valid for another release.
I'm currently trying to finish the missing part (the file editing portion) but I saw your comment https://github.com/joomla/joomla-cms/pull/35911#issuecomment-966319851 and I'm trying to figure a way to satisfy the a11y needs. It would be helpful if I wasn't triggered earlier by someone that doesn't understand that the web evolves and new features cannot always be introduced without B/C breaks. Also unfortunately I haven't been paid, but even if someone offered money for some unethical coding I wouldn't accept it. Maybe that's the reason I'm not rich...

rinka88 commented:
I know this thread is closed. But I would like to know whether such a view will be implemented in the next versions?

dgrammatiko commented:
@rinka88 I can only say that I'm not working on this anymore, for anything related to v5 you have to ask @HLeithner

rinka88 commented:
it would be a pity if such good things were lost. And you can not offer them in the form of pr?

dgrammatiko commented:
> And you can not offer them in the form of pr?

I'm afraid this is not the right way, let me explain: finishing this PR means I have to personally spend many man-hours coding/styling/testing/etc with a huge uncertainty if the PR will ever get merged (if not it was just wasted time). I cannot afford to dedicate the needed time right now, but even I was able to do so, I wouldn't do that before the maintainers gave me a clear signal that this work won't be ignored/turned down...

Anyways, September is a very busy month for me and I think after September there's no reasonable time to make this happen. Maybe in 5.1 or later (if people are not too attached to b/c breaks within a single component) and there is a competent release leader...
